### PR TITLE
optimize interpolation data storage

### PR DESCRIPTION
**Pull request by ashesh2512**
moving away from raw pointers and using std vectors to resize and store data on the fly.

### CODE REVIEW

File: src/CartBlock.C [hunk 1]
```diff
@@ -80,35 +79,22 @@ void CartBlock::getInterpolatedData(int *nints,int *nreals,int **intData,
...
-       for(i=0;i<nrealold;i++) (*realData)[i]=tmpreal[i];      
-       TIOGA_FREE(tmpint);
-       TIOGA_FREE(tmpreal);
-      }
+      intData.resize(3*(*nints));
+      realData.resize(*nreals);
+
       listptr=interpList;
-      icount=3*nintold;
+      icount=3*nintold; icount=3*nintold;
```
Comment (jsitaraman) [inline src/CartBlock.C:90]: statement repeated here ?

**Comment by jsitaraman** [comment]
Can you fix line 90 ? I will merge after that. Otherwise everything looks good

File: src/CartBlock.C (hunk 1 above)
Comment (ashesh2512) [reply src/CartBlock.C]: @jsitaraman Fixed it in `c0718b093c605da25d916f1c9ccb5d8e1d8203aa`

File: src/CartBlock.C [hunk 2]
```diff
@@ -133,15 +119,17 @@ void CartBlock::getInterpolatedData(int *nints,int *nreals,int **intData,
...
-        for(n=0;n<(nvar_cell+nvar_node);n++) (*realData)[dcount++]=qq[n]; // update solution
+        for(n=0;n<(nvar_cell+nvar_node);n++) realData[dcount++]=qq[n]; // update solution
 #endif
 
         listptr=listptr->next;
       }
+
 #ifdef TIOGA_HAS_GPU
-      getInterpolatedDataDevice(&((*realData)[dcount]),nvar_cell,nvar_node);
+      getInterpolatedDataDevice(&((realData.data())[dcount]),nvar_cell,nvar_node);
```
Comment (aprokop) [inline src/CartBlock.C:129]:
```suggestion
      getInterpolatedDataDevice(realData.data() + dcount,nvar_cell,nvar_node);
```

File: src/cartOps.C [hunk 3]
```diff
@@ -473,7 +458,7 @@ void MeshBlock::getInterpolatedSolutionAMR(int *nints,int *nreals,int **intData,
                         m_info_device);
   TIOGA::gpu::synchronize();
 
-  TIOGA::gpu::pull_from_device<double>(&((*realData)[dcount]),realData_d,sizeof(double)*ninterp_total_g*nvar);
+  TIOGA::gpu::pull_from_device<double>(&((realData.data())[dcount]),realData_d,sizeof(double)*ninterp_total_g*nvar);
```
Comment (aprokop) [inline src/cartOps.C:461]: See comment above.

**Pull request merged**